Program.cs

```diff
@@ -88,6 +110,34 @@ internal static async Task<int> Main(string[] args)
 
                 var manifest = (Manifest?) serializer.Deserialize(reader) ?? throw new InvalidDataException();
 
+                bool matching = false;
+                for (int i = 0; i < compareManifests.Count; i++)
+                {
+                    var xmlNode = compareManifests[i] ?? throw new InvalidDataException();

```

Comment:
I'd just use reader.ReadAsync() again for said above thing, just use two readers.